src/ngraph/op/abs.cpp

```diff
@@ -43,3 +46,55 @@ void op::Abs::generate_adjoints(autodiff::Adjoints& adjoints, const OutputVector
 
     adjoints.add_delta(x, delta * make_shared<op::Sign>(x));
 }
+
+namespace
+{
+    template <element::Type_t ET>
+    inline bool evaluate(const HostTensorPtr& arg0, const HostTensorPtr& out, const size_t count)
+    {
+        using T = typename element_type_traits<ET>::value_type;
+        runtime::reference::abs<T>((arg0->get_data_ptr<ET>()), (out->get_data_ptr<ET>()), count);
+        return true;
+    }
+
+    bool evaluate_abs(const HostTensorPtr& arg0, const HostTensorPtr& out, const size_t count)
+    {
+        if (arg0->get_element_type() != out->get_element_type())
```

Comment:
Use `out->set_unary(arg0)`. It takes care of the shape and element type. Your check here is actually wrong because the element type on out is likely to be dynamic at this point.